archunit/src/test/java/com/tngtech/archunit/library/dependencies/GraphTest.java

```diff
@@ -128,24 +132,37 @@ public void complete_graph() {
     @Test
     public void graph_which_causes_error_when_dependently_blocked_nodes_are_not_cleared_after_unblocking() {
         ImmutableSet<Integer> nodes = ImmutableSet.of(0, 1, 2, 3, 4, 5);
-        Graph<Integer, Integer> graph = new Graph<>();
+        Graph<Integer, Object> graph = new Graph<>();
         graph.addNodes(nodes);
 
         graph.addEdges(ImmutableSet.of(
-                new Edge<Integer, Integer>(0, 4),
-                new Edge<Integer, Integer>(1, 0),
-                new Edge<Integer, Integer>(1, 5),
-                new Edge<Integer, Integer>(2, 1),
-                new Edge<Integer, Integer>(3, 1),
-                new Edge<Integer, Integer>(3, 5),
-                new Edge<Integer, Integer>(4, 3),
-                new Edge<Integer, Integer>(5, 1),
-                new Edge<Integer, Integer>(5, 2)
+                newEdge(0, 4),
+                newEdge(1, 0),
+                newEdge(1, 5),
+                newEdge(2, 1),
+                newEdge(3, 1),
+                newEdge(3, 5),
+                newEdge(4, 3),
+                newEdge(5, 1),
+                newEdge(5, 2)
         ));
 
         assertThat(graph.findCycles()).isNotEmpty();
     }
 
+    // This test covers some edge cases, e.g. if too many nodes stay blocked
```

Comment:
This edge case is already covered by `graph_which_causes_error_when_dependently_blocked_nodes_are_not_cleared_after_unblocking`

Reply:
Hmm, to be honest, when I was done I tested with `org.hibernate` and it crashed because I had made some wrong refactoring with respect to unblocking. Yet all existing tests were green. Since I felt like I had already invested much time, I decided to not dig deeper to reduce the problem to a minimal example, but instead just use the Hibernate graph that caused the problem.
So I don't exactly know the difference to the other test, I just know that the way I broke unblocking did not break the existing test :frowning:
I'm still not sure if it's worth it to invest the time though, but yes, maybe the comment should then be adjusted? Or do you think we should dig deeper?

Reply:
The error here was that the whole node was not unblocked though, not that the dependently blocked nodes weren't cleared correctly. So it is also not the same case...

Reply:
So maybe the comment is fine then, because if you carefully read it, the one test says "when dependently blocked nodes are not cleared" while the comment says "if nodes stay blocked". But maybe it could be written clearer?

Reply:
No, I just didn't know the background and didn't get the lack of the word "dependently".